File: sdk/php/gui/charts/UXChart.php
<?php
namespace php\gui\charts;


use php\gui\layout\UXRegion;

class UXChart extends UXRegion{
    /**
    * @var string
    */
    public $title;
    /**
    * @var bool
    */
    public $animated;
    /**
    * @var string
    */
    public $titleSide;
    /**
    * @var bool
    */
    public $legendVisible;
    /**
    * @var string
    */
    public $legendSide;
}

File: src-jvm/main/java/org/develnext/jphp/ext/charts/classes/charts/UXChart.java
package org.develnext.jphp.ext.charts.classes.charts;

import org.develnext.jphp.ext.charts.ChartsExtension;
import org.develnext.jphp.ext.javafx.classes.layout.UXRegion;

import javafx.geometry.Side;
import javafx.scene.chart.Chart;
import php.runtime.annotation.Reflection.Abstract;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Property;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;


@Namespace(ChartsExtension.NS)
@Name("UXChart")
@Abstract
public class UXChart<T extends Chart> extends UXRegion<T>{
	interface WrappedInterface{
		@Property boolean animated();
		@Property Side legendSide();
		@Property boolean legendVisible();
		@Property String title();
		@Property Side titleSide();
	}
	
	public UXChart(Environment env, T wrappedObject){
		super(env, wrappedObject);
	}
	public UXChart(Environment env, ClassEntity clazz){
		super(env, clazz);
	}
}
